bin/build-asset-cdn-json.php

```diff
@@ -1,20 +1,48 @@
 <?php
 
-$path      = dirname( dirname( __FILE__ ) ) . '/';
+// The repo root path.
+$path = dirname( dirname( __FILE__ ) ) . '/';
+
+// Build an iterator over all files in the repo that match the regex in the RegexIterator.
 $directory = new RecursiveDirectoryIterator( $path );
 $iterator  = new RecursiveIteratorIterator( $directory );
 $regex     = new RegexIterator( $iterator, '/^.+\.(css|js)$/i', RecursiveRegexIterator::GET_MATCH );
 
+$ignore_paths = array(
+	'_inc/client/',
+	'bin/',
+	'docker/',
+	'docs/',
+	'extensions/',
+	'logs/',
+	'node_modules/',
+	'tests/',
+	'tools/',
+	'vendor/',
+);
+
 $manifest = array();
-foreach ( $regex as $file => $value ) {
-	$file = str_replace( $path, '', $file );
-	$directory = substr( $file, 0, strpos( $file, '/' ) );
-	if ( in_array( $directory, array( 'node_modules', 'tests' ) ) ) {
+foreach ( $regex as $path_to_file => $value ) {
+	$path_from_repo_root = str_replace( $path, '', $path_to_file );
+
+	// Ignore top-level files.
+	if ( false === strpos( $path_from_repo_root, '/' ) ) {
 		continue;
 	}
-	$manifest[] = $file;
+
+	// Ignore explicit ignore list.
+	foreach ( $ignore_paths as $ignore_path ) {
+		if ( 0 === strpos( $path_from_repo_root, $ignore_path ) ) {
+			continue 2;
+		}
+	}
+
+	$manifest[] = $path_from_repo_root;
 }
 
 $export = var_export( $manifest, true );
 
-file_put_contents( $path . 'modules/photon-cdn/jetpack-manifest.php', "<?php \r\n\$assets = $export;\r\n" );
+file_put_contents( $path . 'modules/photon-cdn/jetpack-manifest.php', "<?php
+// This file is autogenerated by bin/build-asset-cdn-json.php
+
+$assets = $export;\r\n" );
```

Comment:
This gives a

```
PHP Notice:  Undefined variable: assets in ~/jetpack/bin/build-asset-cdn-json.php on line 48
```

Can we suppress that somehow?

Reply:
E.g. using a [`HEREDOC` vor `NOWDOC`](https://stackoverflow.com/a/1848974)? (those names! :joy: )

Reply:
Oh, the output seems actually broken right now:

```php
<?php
// This file is autogenerated by bin/build-asset-cdn-json.php

 = array (
  0 => 'css/jetpack-icons.min.css',
  1 => 'css/jetpack-icons-rtl.css',
  2 => 'css/jetpack-idc-admin-bar.min.css',
  3 => 'css/jetpack-admin-rtl.css',
/* etc */
```

Reply:
Whoops… I'll grab that…

Reply:
f02923305501c38564fa55a2cab08cdff0205585